shared/prow/prow_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// prow_test.go contains unit tests for prow package
+
+package prow
+
+import (
+	"testing"
+	"path"
+	"os"
+)
+
+const(
+	InvalidJobType = "invalid"
+)
+
+func TestGetArtifacts(t *testing.T) {
+	dir := os.Getenv("ARTIFACTS")
+
+	// Test we can read from the env var
+	os.Setenv("ARTIFACTS", "test")
+	v := GetLocalArtifactsDir()
+	if v != "test" {
+		t.Fatalf("Actual artifacts dir: '%s' and Expected: 'test'", v)
+	}
+
+	// Test we can use the default
+	os.Setenv("ARTIFACTS", "")
+	v = GetLocalArtifactsDir()
+	if v != "artifacts" {
+		t.Fatalf("Actual artifacts dir: '%s' and Expected: 'artifacts'", v)
+	}
+
+	// Set it to originial value
+	os.Setenv("ARTIFACTS", dir)
+}
+
+func TestPeriodicJobPath(t *testing.T) {
+	job := Job{
+		Name: "job_0",
+		Type: PeriodicJob,
+	}
+	expectedPath := path.Join("logs", job.Name)
+	if actualPath := job.GetPath(); actualPath != expectedPath {
+		t.Fatalf("Exptect '%s', actual '%s'", expectedPath, actualPath)
+	}
+}
+
+func TestPostsubmitJobPath(t *testing.T) {
+	job := Job{
+		Name: "job_0",
+		Type: PostsubmitJob,
+	}
+	expectedPath := path.Join("logs", job.Name)
+	if actualPath := job.GetPath(); actualPath != expectedPath {
+		t.Fatalf("Exptect '%s', actual '%s'", expectedPath, actualPath)
+	}
+}
+
+func TestPresubmitJobPath(t *testing.T) {
+	job := Job{
+		Name: "job_0",
+		Type: PresubmitJob,
+	}
+	
+	expectedPath := path.Join("pr-logs/pull/", OrgName + "-" + "", "", job.Name)
+	if actualPath := job.GetPath(); actualPath != expectedPath {
+		t.Fatalf("Exptect '%s', actual '%s'", expectedPath, actualPath)
+	}
+}
+
+func TestBatchJobPath(t *testing.T) {
```

Comment:
Replace the 4 separate and almost identical `Test*JobPath` tests with a single, data-driven test.

Reply:
done